.github/workflows/macos_unit_tests.yaml

```
@@ -0,0 +1,54 @@
+name: macos tests
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+    branches:
+      - master
+      - develop
+jobs:
+  build:
+
+    runs-on: macos-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - uses: actions/checkout@v1
```

Use already the newer version, v1 had some issues recently.
```suggestion
    - uses: actions/checkout@v2
```